pkg/webhook/kafkacluster_validator_test.go

```diff
@@ -0,0 +1,341 @@
+							{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default",
+					},
+				},
+			},
+			isValid: true,
+		},
+		{
+			testName: "2",
+			kafkaClusterSpecNew: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+					"default2": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							//	v1beta1.StorageConfig{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default2",
+					},
+				},
+			},
+			kafkaClusterSpecOld: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			testName: "3",
+			kafkaClusterSpecNew: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							//v1beta1.StorageConfig{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default",
+					},
+				},
+			},
+			kafkaClusterSpecOld: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default",
+					},
+				},
+			},
+			isValid: false,
+		},
+		{
+			testName: "4",
+			kafkaClusterSpecNew: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							{MountPath: "logs1"},
+							{MountPath: "logs2"},
+							{MountPath: "logs3"},
+						},
+					},
+				},
+				Brokers: []v1beta1.Broker{
+					{
+						Id:                1,
+						BrokerConfigGroup: "default",
+					},
+				},
+			},
+			kafkaClusterSpecOld: v1beta1.KafkaClusterSpec{
+				BrokerConfigGroups: map[string]v1beta1.BrokerConfig{
+					"default": {
+						StorageConfigs: []v1beta1.StorageConfig{
+							//v1beta1.StorageConfig{MountPath: "logs1"},
```

Comment:
**Note**: unnecessary comment? (unless it is to understand the case more easily)

Reply:
It can be removed, but you can see also more easily that this one has been removed from the new Config. So this validation should be rejected

Reply:
If we wish to show that should we not have a comment indicating that instead? Like instead of this you could write in the comment that the removed MountPath is this.

Reply:
I agree with @Kuves